### PR TITLE
python3Packages.ibis-framework: add extras-require to passthru

### PR DESCRIPTION
**Pull request by cpcloud**
###### Description of changes

This PR adds the ability to select specific features of the ibis-framework package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cpcloud** [comment]
@SuperSandro2000 Any thoughts here?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please use https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#optional-extra-dependencies

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 1]
```diff
@@ -136,12 +136,27 @@ buildPythonPackage rec {
...
     license = licenses.asl20;
+    passthru = {
+      extras-require = {
+        clickhouse = [ clickhouse-cityhash clickhouse-driver lz4 ];
+        dask = [ dask pyarrow ];
+        datafusion = [ datafusion ];
+        duckdb = [ duckdb duckdb-engine sqlalchemy ];
+        geospatial = [ geoalchemy2 geopandas shapely ];
+        mysql = [ pymysql sqlalchemy ];
+        pandas = [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ibis-framework/default.nix:153]: Not sure if we need this when it is empty.

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 2]
```diff
@@ -136,12 +136,27 @@ buildPythonPackage rec {
...
   pythonImportsCheck = [
     "ibis"
-  ] ++ (map (backend: "ibis.backends.${backend}") backends);
+  ] ++ map (backend: "ibis.backends.${backend}") testBackends;
 
   meta = with lib; {
     description = "Productivity-centric Python Big Data Framework";
     homepage = "https://github.com/ibis-project/ibis";
     license = licenses.asl20;
+    passthru = {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ibis-framework/default.nix:145]: passthru goes onto the same level as meta, not into meta.

File: pkgs/development/python-modules/ibis-framework/default.nix (hunk 1 above)
Comment (cpcloud) [reply pkgs/development/python-modules/ibis-framework/default.nix]: It's a feature upstream and used in constructing the test backend list here.

File: pkgs/development/python-modules/ibis-framework/default.nix (hunk 2 above)
Comment (cpcloud) [reply pkgs/development/python-modules/ibis-framework/default.nix]: Fixed, thanks!

File: pkgs/development/python-modules/ibis-framework/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/ibis-framework/default.nix]: I can't find it in https://github.com/ibis-project/ibis/blob/3.0.2/setup.py#L68-L134 . Am I looking at the right repo?

Comment (cpcloud) [reply pkgs/development/python-modules/ibis-framework/default.nix]: Right repo, wrong file 😄: https://github.com/ibis-project/ibis/blob/3.0.2/pyproject.toml#L143

File: pkgs/development/python-modules/ibis-framework/default.nix [hunk 3]
```diff
@@ -136,12 +136,28 @@ buildPythonPackage rec {
...
+  ] ++ map (backend: "ibis.backends.${backend}") testBackends;
 
   meta = with lib; {
     description = "Productivity-centric Python Big Data Framework";
     homepage = "https://github.com/ibis-project/ibis";
     license = licenses.asl20;
     maintainers = with maintainers; [ costrouc cpcloud ];
   };
+
+  passthru = {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ibis-framework/default.nix:148]: passthru goes right before meta.

**Comment by SuperSandro2000** [comment]
after that we can merge it

**Comment by cpcloud** [comment]
@SuperSandro2000 Done! Thanks for the review.

File: pkgs/development/python-modules/ibis-framework/default.nix (hunk 3 above)
Comment (cpcloud) [reply pkgs/development/python-modules/ibis-framework/default.nix:141]: Done.

**Pull request merged**